src/gitops/components/appset/graph/ApplicationSetGraphView.tsx

```
@@ -324,6 +348,104 @@ export const TreeViewLayout = {
+  selectedIds,
+  controller,
+  applicationSet,
+}: TopologySideBarParams): React.ReactElement => {
+  const selectedNode =
+    selectedIds.length > 0
+      ? (controller.getNodeById(selectedIds[0]) as TopologyNode | undefined)
+      : undefined;
+
+  const isStepGroup = selectedNode?.getId().endsWith('-step-group');
+
+  const pages = React.useMemo(
```

ESLint

[error] React Hook "React.useMemo" is called in function "getTopologySideBar" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

(react-hooks/rules-of-hooks)